### PR TITLE
Update test_python_dispatch.py

### LINKED ISSUE #62501: test_python_dispatch fails under pytest
## 🐛 Bug
```
--- Logging error ---
Traceback (most recent call last):
  File "/scratch/rzou/pt/workspace-env/lib/python3.7/logging/__init__.py", line 1025, in emit
    msg = self.format(record)
  File "/scratch/rzou/pt/workspace-env/lib/python3.7/logging/__init__.py", line 869, in format
    return fmt.format(record)
  File "/scratch/rzou/pt/workspace-env/lib/python3.7/site-packages/_pytest/logging.py", line 65, in format
    return super().format(record)
  File "/scratch/rzou/pt/workspace-env/lib/python3.7/logging/__init__.py", line 608, in format
    record.message = record.getMessage()
  File "/scratch/rzou/pt/workspace-env/lib/python3.7/logging/__init__.py", line 369, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/raid/rzou/pt/workspace-env/bin/pytest", line 11, in <module>
```

## To Reproduce

`pytest test/test_python_dispatch.py -v`

### PR DESCRIPTION
**Pull request by Krystal225**
Fixes #62501 

But now new issue 
```
IndexError: tuple index out of range
```

to reproduce:
```
pytest test/test_python_dispatch.py -v
```

### CODE REVIEW

File: test/test_python_dispatch.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def emit(self, record):
         self.log_list.append(f'{fmt_rets} = {record.msg}{fmt_args}')
 
 def log_input(name: str, var: object):
-    logging.getLogger("LoggingTensor").info("input", (name,), (var,))
+    logging.getLogger("LoggingTensor").info("input {} {}".format(name, var))
```
Comment (zou3519) [inline test/test_python_dispatch.py:86]: I tried this out locally and it doesn't seem to solve the pytest issue. Do you have other ideas on how to fix this?

Using `python test/test_python_dispatch` I am getting:
```
======================================================================
ERROR: test_basic (__main__.TestPythonDispatch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/raid/rzou/pt/fix_tensor_ctor_grad/test/test_python_dispatch.py", line 109, in test_basic
    log_input("x", x)
  File "/raid/rzou/pt/fix_tensor_ctor_grad/test/test_python_dispatch.py", line 91, in log_input
    logging.getLogger("LoggingTensor").info("input {} {}".format(name, var))
  File "/raid/rzou/pt/fix_tensor_ctor_grad-env/lib/python3.9/logging/__init__.py", line 1446, in info
    self._log(INFO, msg, args, **kwargs)
  File "/raid/rzou/pt/fix_tensor_ctor_grad-env/lib/python3.9/logging/__init__.py", line 1589, in _log
    self.handle(record)
  File "/raid/rzou/pt/fix_tensor_ctor_grad-env/lib/python3.9/logging/__init__.py", line 1599, in handle
    self.callHandlers(record)
  File "/raid/rzou/pt/fix_tensor_ctor_grad-env/lib/python3.9/logging/__init__.py", line 1661, in callHandlers
    hdlr.handle(record)
  File "/raid/rzou/pt/fix_tensor_ctor_grad-env/lib/python3.9/logging/__init__.py", line 952, in handle
    self.emit(record)
  File "/raid/rzou/pt/fix_tensor_ctor_grad/test/test_python_dispatch.py", line 85, in emit
    fmt_args = "(" + ", ".join(self._fmt(a) for a in record.args[0]) + ")"
IndexError: tuple index out of range
```
Using `pytest test/test_dispatch.py` I am getting:
```
___________________________________________________ TestPythonDispatch.test_basic ___________________________
Traceback (most recent call last):
  File "/raid/rzou/pt/fix_tensor_ctor_grad/test/test_python_dispatch.py", line 109, in test_basic
    log_input("x", x)
  File "/raid/rzou/pt/fix_tensor_ctor_grad/test/test_python_dispatch.py", line 91, in log_input
    logging.getLogger("LoggingTensor").info("input {} {}".format(name, var))
  File "/raid/rzou/pt/fix_tensor_ctor_grad-env/lib/python3.9/logging/__init__.py", line 1446, in info
    self._log(INFO, msg, args, **kwargs)
  File "/raid/rzou/pt/fix_tensor_ctor_grad-env/lib/python3.9/logging/__init__.py", line 1589, in _log
    self.handle(record)
  File "/raid/rzou/pt/fix_tensor_ctor_grad-env/lib/python3.9/logging/__init__.py", line 1599, in handle
    self.callHandlers(record)
  File "/raid/rzou/pt/fix_tensor_ctor_grad-env/lib/python3.9/logging/__init__.py", line 1661, in callHandlers
    hdlr.handle(record)
  File "/raid/rzou/pt/fix_tensor_ctor_grad-env/lib/python3.9/logging/__init__.py", line 952, in handle
    self.emit(record)
  File "/raid/rzou/pt/fix_tensor_ctor_grad/test/test_python_dispatch.py", line 85, in emit
    fmt_args = "(" + ", ".join(self._fmt(a) for a in record.args[0]) + ")"
IndexError: tuple index out of range
```

**Review by zou3519 — CHANGES_REQUESTED** [review]

File: test/test_python_dispatch.py (hunk 1 above)
Comment (Krystal225) [reply test/test_python_dispatch.py:86]: I am not able to fix this issue. if you have any suggestions please tell and I will write the code. 
Thankyou

Comment (Harshshokeen09) [reply test/test_python_dispatch.py:86]: I have made a pull request kindly see to it please
It is solving the issue

Comment (Harshshokeen09) [reply test/test_python_dispatch.py:86]: Need anything more?
Rather than my PR?

**Comment by zou3519** [comment]
We've already fixed https://github.com/pytorch/pytorch/issues/62501

**Pull request closed without merging**